pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -406,7 +406,32 @@ func (a *ApplicationRestoreController) handle(ctx context.Context, restore *stor
 	}
 	switch restore.Status.Stage {
 	case storkapi.ApplicationRestoreStageInitial:
-		// Make sure the namespaces exist
+		if restore.Spec.BackupObjectType == resourcecollector.PxBackupObjectType_virtualMachine {
+			// The backuObjectType is VirtualMachine. We need to remove resources such as PVC,
+			// configMap, secret that are not associated to any VirtualMachine (aka orphaned resources).
```

Comment:
" secret that are not associated to any VirtualMachine " --> would suggest " secret that are not associated to any VirtualMachine selected for restore". Because rest of the resources are also belongs to the VM But they are not selected for restore in the current restore.